src/browser/base/content/browser-js.patch

```diff
@@ -10,7 +10,30 @@ index 32b67f846b9942ad3da4276bb8748fee915eb485..529ef8895d2ab01b4cbebd2992fd3b64
    DevToolsSocketStatus:
      "resource://devtools/shared/security/DevToolsSocketStatus.sys.mjs",
    DownloadUtils: "resource://gre/modules/DownloadUtils.sys.mjs",
-@@ -2293,6 +2294,8 @@ var XULBrowserWindow = {
+@@ -820,12 +821,21 @@ function UpdateBackForwardCommands(aWebNavigation) {
+ 
+   var backDisabled = backCommand.hasAttribute("disabled");
+   var forwardDisabled = forwardCommand.hasAttribute("disabled");
++  
++  // Check if we should keep back button enabled for regular unpinned tabs
++  var shouldEnableBackForTabClose = false;
++  if (!aWebNavigation.canGoBack) {
++    shouldEnableBackForTabClose = gZenCommonActions.shouldCloseTabOnBack(gBrowser?.selectedTab);
++  }
++  
+   if (backDisabled == aWebNavigation.canGoBack) {
+     if (backDisabled) {
```

Comment:
Shouldn't we set `backDisabled` instead of making a new variable `shouldEnableBackForTabClose` here?